File: ObjC/Unit.m
//
//  Unit.m
//  Pods
//
//  Created by materik on 10/02/16.
//
//

#import "Unit.h"

@implementation Unit

- (instancetype)initWithValue:(float)value {
    self = [super init];
    if (self) {
        [self setValue:value];
    }
    return self;
}

- (NSString *)description {
    return [self formatted];
}

- (NSString *)unit {
    return @"";
}

- (NSString *)formatted {
    return [self formattedWithPrecision:2];
}

- (NSString *)formattedWithPrecision:(NSUInteger)precision {
    NSString *format = [NSString stringWithFormat:@"%%.%ldf", (long)precision];
    NSString *value = [NSString stringWithFormat:format, self.value];
    return [NSString stringWithFormat:@"%@ %@", value, self.unit];
}

+ (instancetype)value:(float)value {
    return [[[self class] alloc] initWithValue:value];
}

@end

File: Swift/Unit.swift
//
//  Unit.swift
//  Pods
//
//  Created by materik on 16/02/16.
//
//

import Foundation

public class Unit: CustomStringConvertible {
    
    public var value: Float = 0.0
    public var description: String { return self.formatted() }
    internal var unit: String { return "" }
    
    public required init(_ value: Float) {
        self.value = value
    }
    
    public func formatted(withPrecision precision: Int = 2) -> String {
        let format = String(format: "%%.%df", precision)
        let value = String(format: format, self.value)
        return "\(value) \(self.unit)"
    }
    
}
